cassovary-core/src/main/scala/com/twitter/cassovary/graph/bipartite/BipartiteGraph.scala

```diff
@@ -153,18 +168,24 @@ class BipartiteGraph(val leftNodes: Array[BipartiteNode], val leftNodeCount: Int
 
     if (id == 0) {
       return None // id 0 is not allowed in bipartite graph
-    } else if (isLeftNodeId) {
-      //invert nodeId to index
-      val leftOriginalId = - id
-      if (leftOriginalId < leftNodes.length) {
-        if (leftNodes(leftOriginalId) != null) Some(leftNodes(leftOriginalId))
-        else None
-      } else None
+    } else if (!nodeIdsUnique) {
+      if (isLeftNodeId) {
+        //invert nodeId to index
+        val leftOriginalId = -id
+        if (leftOriginalId < leftNodes.length) {
```

Comment:
refactor this code segment into a convenience method, as the same code is used 4 times below.

if (id < arr.length && arr(id) != null) Some(arr(id)) else None